UnitTest/tests/test_sub_SIGNALduino_Get-definition.txt

```diff
@@ -181,6 +181,91 @@ defmod test_sub_SIGNALduino_Get UnitTest dummyDuino
 		$SD_Parse->unmock;
 	}; 
 
+	subtest 'Test device with dummy attrib 1' => sub {
+		plan tests => 3;
 
+		subtest 'Test Attributes (hardware undef)' => sub {
+			plan tests => 2;
+			is(AttrVal($target, "dummy", 0),1,"check attrib dummy is 1");
+			delete $attr{$target}{hardware} if (AttrVal($target, "hardware", undef));
+			is(AttrVal($target, "hardware", undef),undef,"check attrib hardware undef");
+		};
+
+		subtest 'Test allowed - get commands' => sub {
+			my $rawArg="?";
+			my $ret=SIGNALduino_Get($targetHash, $target, "?", $rawArg);
+
+			my @tests = split(" ", substr($ret,index($ret,"one of")+7));
```

Comment:
Was genau möchtest Du hier extrahieren. Ich verstehe es irgendwie nicht. Die +7 irritierern mich :(
Ich Wüsste auch nicht, wieso die Anzahl der Einträge in @tests die Anzahl der Tests ergibt. Da gibt es keinen Zusammenhang oder doch?

Reply:
Es ist ganz easy.
Man erhält im FHEMWeb folgende Ausgabe als Beispiel

`Unknown argument ?, choose one of availableFirmware:noArg cmds:noArg config:noArg freeram:noArg ping:noArg raw uptime:noArg version:noArg`

Davon suche ich mir die Position von "one of " (mit dem letzten Leerzeichen) und ab dieser Postion habe ich bis zum Ende die möglichen Argumente.

Bsp: `availableFirmware:noArg cmds:noArg config:noArg freeram:noArg ping:noArg raw uptime:noArg version:noArg`

Diese Splitte ich mit einem Leerzeichen und schon habe ich die Anzahl der Argumente. Diese Anzahl ist an derzeitgen Tests definiert. Sollte aber die Anzahl größer oder weniger werden, so meckert der Test *smile* Somit ist sichergestellt, wenn ein Zusätzliches Kommando kommt, das dort der Test angepasst werden muss.

Reply:
Bisher hatten wir für Dummy folgendes als Korrekte Rückgabe:

`check_set match qr/^Unknown argument \?, choose one of availableFirmware:noArg raw$/,!match qr/cc/`

Bis auf, dass kein cc enthalten sein darf sind die sehr ähnlich :)

Reply:
Ähnlich ja, aber ich zähle diese auf und ich sage nocheinmal, mit dieser Prüfung testern wir nur die Verfügbarkeit und nicht die reine Funktion.

Zusätzlich wurde auch jedes derzeitige Kommando aufgelistet weil dein Ansinnen war, nichts unter den Tisch fallen zu lassen. (ausgenommen wie du schon feststelltest die cc1101 ...)

Reply:
Ah, ich denke ich habe es verstanden. War für mich erst mal ein wenig "komisch", weil Du nicht einen Test auf die Anzahl an Einträge macht, sonder die Anzahl an Einträgen mit der Anzahl der Tests gleichsetzt, diese Menge aber als Indikator verwendest, dass ein Test notwendig ist.

Die Regex würde allerdings auch immer felschlagen, wenn jemand ein weiteres Kommando hinzufügt, da sowohl Start als auch Ende markiert sind :)
`^Unknown argument \?, choose one of availableFirmware:noArg raw$`